build_all/base-libraries/AzureIoTHub/src/scripts/automate_board_config.py

```diff
@@ -0,0 +1,200 @@
+                file_modified = True
+            else:
+                line = line.rstrip()
+                line = f"{line}{str_append}\n"
+                file_modified = True
+        sys.stdout.write(line)
+    return file_modified
+
+
+def confirm_overwrite(file_path):
+    '''
+    Confirms whether to overwrite changes otherwise exits program
+
+    :param file_path: The path to the file
+    :type file_path: str
+    '''
+    prompt = f"There is already a backup file at" \
+             f" {file_path}; proceeding will" \
+             f" overwrite this file. Do you wish to proceed?" \
+             f" Input Y or N:" \
+             f" "
+    while True:
+        response = input(prompt)
+        response = response.lower()
+        if response == 'n':
+            print("No changes made... exiting")
+            sys.exit()
+        elif response == 'y':
+            print("Backup will be overwritten")
+            break
+        else:
+            print("Ensure your response is a Y or N")
+
+
+def main():
+    disclaimer_prompt = \
+        "This script will attempt to automatically update" \
+        " your ESP8266 and/or ESP32 board files to work with Azure IoT Hub" \
+        " for the repo https://github.com/Azure/azure-iot-arduino" \
+        "\nPlease refer to the license agreement there." \
+        "\nThis script will update all installed versions of board" \
+        " libraries for ESP8266 and/or ESP32." \
+        "\nDo you wish to proceed? Please answer Y or N:" \
+        " "
+
+    while True:
+        response = input(disclaimer_prompt)
+        response = response.lower()
+        if response == 'n':
+            print("No changes made... exiting")
+            sys.exit()
+        elif response == 'y':
+            print("Proceeding")
+            break
+        else:
+            print("Ensure your response is a Y or N")
+
+    board_prompt = \
+        "Would you like to update your ESP8266 or ESP32 board files?\n" \
+        "For ESP8266 please respond: 8266\n" \
+        "For ESP32 please respond: 32\n" \
+        "Which board files would you like to update:" \
+        " "
+
+    board_to_update = ""
+    PACKAGE_PATH = ""
+    while True:
+        response = input(board_prompt)
+        response = response.lower()
+        if response == '8266':
+            board_to_update = '8266'
+            PACKAGE_PATH = ESP8266_PACKAGE_PATH
+            break
+        elif response == '32':
+            board_to_update = '32'
+            PACKAGE_PATH = ESP32_PACKAGE_PATH
+            break
+        else:
+            print("Ensure your response is either 8226 or 32")
+
+    if sys.platform == "darwin":
+        ARDUINO_PACKAGES_PATH = Path(Path.home() / "Library/Arduino15")
+    elif sys.platform == "linux":
+        ARDUINO_PACKAGES_PATH = Path(Path.home() / ".arduino15/packages/")
+    elif sys.platform == "win32":
+        ARDUINO_PACKAGES_PATH = Path(Path.home() / "AppData/Local/Arduino15")
```

Comment:
On my machine, my Arduino stuff was installed to a non-standard location. We could prompt the user here if the resolution of this fails?

Reply:
@danewalton So this packages path is different from your libraries path. This is not one that the vast majority of people change, and is not even changeable in the Preferences window.

Reply:
I think regardless, it's a hiccup that I hit which if someone else hits, would leave them DIW. What we're hoping for here is a solution which is resilient to hiccups which provides us with less Github issues of someone saying "This doesn't work for me it says it can't find this path". Also, if someone has to hardcode this, it changes the git diff and leads to muddying of work in progress.

Reply:
I agree. It can be added as a parameter that a user can enter, and a note/hint can be added if the path is not found. @codycodes see [parse_opts](https://github.com/Azure/azure-iot-sdk-c/blob/master/testtools/UART_interface/serial_connect.py#L33) for how some of our python code ingests parameters when run in the command line.

Reply:
Looking at that snippet I realized an error which wasn't discovered until I ran the script on a Linux instance I spun up; this didn't crop up because previously we didn't test on Linux (only macOS and Windows); for cross-platform things like this I'll ensure test for those in the future. I've fixed the packages path and tested it and it's working.

I'll take into account those changes using the specified `getopt.gnu_getopt(args, shortopts, longopts=[])` function and have commits ready soon.

I also realize there could be confusion around the generic error message if the board path isn't found, so I will include the location the script is looking in the error output so it's more apparent if it fails on an incorrect path.

The script also only tests for Linux/Windows/macOS and otherwise prints that it couldn't set the path; should other operating systems be considered? I'm happy to test for those as well.

In any case if this script doesn't work for whatever reason, I've kept the documentation as-is so that the original, manual modification of files can be followed as a fallback.

Reply:
I would also look to use https://docs.python.org/3.8/library/argparse.html#module-argparse for options parsing. `getopt` and `optparse` seems to be deprecated so it would be good for us to move to the latest.

Reply:
I've added the command line parameters following the example from @jbobotek.

I also added verbosity to the error output to help users understand why the script couldn't locate package path.